app/liquid_tags/github_tag/github_readme_tag.rb

```diff
@@ -3,71 +3,79 @@
 class GithubTag
   class GithubReadmeTag
     PARTIAL = "liquids/github_readme".freeze
+    GITHUB_DOMAIN_REGEXP = %r{.*github.com/}.freeze
+    OPTION_NO_README = "no-readme".freeze
+    VALID_OPTIONS = [OPTION_NO_README].freeze
 
-    attr_reader :client, :content, :options, :readme_html
-
-    def initialize(link)
-      parsed_link = parse_link(link)
-      @options = parse_options(link)
-      @client = Octokit::Client.new(access_token: token)
-      @content = client.repository(parsed_link)
-      @readme_html = fetch_readme(parsed_link)
+    def initialize(input)
+      @repository_path, @options = parse_input(input)
     end
 
     def render
+      content = Github::Client.repository(repository_path)
+
+      if show_readme?
+        readme_html = fetch_readme(repository_path)
+      end
+
       ActionController::Base.new.render_to_string(
         partial: PARTIAL,
         locals: {
           content: content,
-          show_readme: show_readme? && readme_html.present?,
+          show_readme: readme_html.present?,
           readme_html: readme_html
         },
       )
+    rescue Github::Errors::NotFound, Github::Errors::InvalidRepository
+      raise_error
     end
 
     private
 
-    def parse_link(link)
-      link = sanitize_link(link)
-      parsed_link = link.split(" ").first.delete(" ")
-      raise_error if parsed_link.split("/").length > 2
+    attr_reader :repository_path, :options, :content
 
-      parsed_link
-    end
+    def parse_input(input)
+      sanitized_input = sanitize_input(input)
 
-    def valid_option(option)
-      option.match(/no-readme/)
-    end
+      path, *options = sanitized_input.split(" ")
 
-    def parse_options(link)
-      opts = sanitize_link(link)
-      _, *options = opts.split(" ")
+      validate_options!(*options)
 
-      validated_options = options.map { |option| valid_option(option) }.reject(&:nil?)
-      raise StandardError, "GitHub tag: `#{link}`: Invalid option - did you mean `no-readme`?" unless options.empty? || validated_options.any?
+      path.gsub!(%r{/\z}, "") # remove optional trailing forward slash
+      repository_path = URI.parse(path)
+      repository_path.query = repository_path.fragment = nil
```

Comment:
Hm, I'm unfamiliar this syntax; can you explain how we're deciding what to assign `repository_path.query` to here? I feel like I'm about to learn a new Ruby thing

Reply:
If path contains query parameters or fragments, `path.query` and `path.fragment` will be populated:

```ruby
[2] pry(main)> path.fragment
=> "issue?s=query"
[3] pry(main)> path = URI.parse("/path?s=query#fragment")
=> #<URI::Generic /path?s=query#fragment>
[4] pry(main)> path.fragment
=> "fragment"
[5] pry(main)> path.query
=> "s=query"
[6] pry(main)> path.to_s
=> "/path?s=query#fragment"
```

since we don't need neither parameters not the fragment we just remove them, so that the string version of the path will be devoid of them:

```ruby
[7] pry(main)> path.query = path.fragment = nil
=> nil
[8] pry(main)> path.to_s
=> "/path"
```